### PR TITLE
Clone Schedule Functionality

### LINKED ISSUE #128: Have way to copy schedule between schedule versions
Having some way to copy a schedule version in its entirety or copy some courses to another schedule version would be nice. Alternately, a duplicate schedule version button might also solve this.

### PR DESCRIPTION
**Pull request by adityavidyadharan**
Resolves #128 

Adds a button to clone a schedule in its current state

Screen Shot 2022-08-30 at 1 10 00 AM

### CODE REVIEW

File: src/components/HeaderDisplay/index.tsx [hunk 1]
```diff
@@ -202,6 +204,20 @@ function VersionSelector({ state }: VersionSelectorProps): React.ReactElement {
...
+          // Add the duplicate button
+          actions.push({
+            type: 'button',
+            icon: faCopy,
+            onClick: () => {
+              state.cloneVersion(
+                version.id,
+                getNextVersionName(
+                  state.allVersionNames.map(({ name }) => name)
+                )
```
Comment (jazeved0) [inline src/components/HeaderDisplay/index.tsx:216]: Thoughts on using this instead? I think it makes it clearer what the new schedule is, though feel free to disagree.

```suggestion
                `Copy of ${version.name}`
```

File: src/data/hooks/useVersionActions.ts [hunk 2]
```diff
@@ -147,5 +149,36 @@ export default function useVersionActions({
     [updateTermScheduleData]
   );
 
-  return { addNewVersion, deleteVersion, renameVersion };
+  const cloneVersion = useCallback(
```
Comment (jazeved0) [inline src/data/hooks/useVersionActions.ts:152]: 👍 nice clean implementation.

Initially, I was slightly concerned that having two calls to `updateTermScheduleData` (one inside `addNewVersion`, and one here) might cause two separate renders if they aren't batched, but I just checked it locally and it only re-renders once in the React DevTools Profiler. No problems here 👍

**Review by jazeved0 — CHANGES_REQUESTED** [review]
Very minor suggested changes, overall great 👍

File: src/components/HeaderDisplay/index.tsx (hunk 1 above)
Comment (adityavidyadharan) [reply src/components/HeaderDisplay/index.tsx:216]: Sounds good

**Review by jazeved0 — APPROVED** [review]
Looks good to me! :shipit:

**Pull request merged**